src/pages/instructor/Instructor.vue

```diff
@@ -5,39 +5,64 @@
 <script lang="ts">
 import { defineComponent } from 'vue';
 import { useRoute } from 'vue-router';
-import { useStore } from 'src/store';
 import { setI18nLanguage } from 'boot/i18n';
+import { logger } from 'boot/logger';
+
+import { useUserStore } from 'src/stores/users';
+import { useSettingsStore } from 'src/stores/settings';
+import { useProblemSetStore } from 'src/stores/problem_sets';
+import { useSessionStore } from 'src/stores/session';
+import { parseRouteCourseID } from 'src/router/utils';
 
 export default defineComponent({
 	name: 'Instructor',
 	props: {
 		course_name: String,
 		course_id: String
 	},
-	setup(props) {
-		const store = useStore();
-
-		if (props.course_id && props.course_name) {
-			const course = {
-				course_id: props.course_id,
-				course_name: props.course_name
-			};
-			void store.dispatch('session/setCourse', course);
+	setup() {
+		const session = useSessionStore();
+		const users = useUserStore();
+		const route = useRoute();
+
+		const course_id = parseRouteCourseID(route);
+		const course = users.user_courses.find(c => c.course_id === course_id);
+		if (course) {
+			void session.setCourse({
+				course_id,
+				course_name: course.course_name
+			});
 		}
 	},
-	async created() {
+	created() {
 		// fetch most data needed for instructor views
-		const store = useStore();
+		const users = useUserStore();
+		const settings = useSettingsStore();
+		const problem_sets = useProblemSetStore();
 		const route = useRoute();
-		await store.dispatch('users/fetchMergedUsers', route.params.course_id);
-		await store.dispatch('problem_sets/fetchProblemSets', route.params.course_id);
-		await store.dispatch('problem_sets/fetchSetProblems', route.params.course_id);
-		await store.dispatch('settings/fetchDefaultSettings');
-		await store.dispatch('settings/fetchCourseSettings', route.params.course_id);
-
-		// Set the language from the course settings.
-		await setI18nLanguage((store.getters as { [key: string]: (var_name: string) => string })
-			['settings/get_setting_value']('language'));
+
+		const course_id = parseRouteCourseID(route);
+
+		logger.debug('[Intructor]: fetching users from the server.');
+		void users.fetchMergedUsers(course_id);
+
+		logger.debug('[Instructor]: fetch problem_sets from server');
+		void problem_sets.fetchProblemSets(course_id);
+
+		void problem_sets.fetchSetProblems(course_id);
```

Comment:
I guess this will do for now, but this really shouldn't be done here.  The problems shouldn't be fetched until a set is actually loaded, and at that time only the problems for that set should be loaded.  Not all problems for all sets.  In fact, the course users and problem sets shouldn't even be loaded at this point.  None of that is needed at this point.  This all needs to be rethought.

On another note, this revealed another regression.  The answer prefix is clearly not being set (or not set uniquely) for the problems when they are loaded for problem sets as the same thing is happening there as was happening in the library browser.  Mathquill answer blanks are all begin created in the first problem.

Reply:
Agreed... more granularity WRT loading data.

I'll have a look and see about the problem prefixes with @pstaabp's latest commit.

Reply:
I was thinking that we load data at the beginning if it is needed for multiple views.  

For example, the reason for loading the set problems is that clearly they are needed for set details, but also, I was thinking that the Problem Sets Manager (editor) should show the number of problems in a given set and therefore would be needed there.   There's work to be done on the Problem Sets Manager.

Another approach would be to wait until the data is needed and then load it (either in the store or just locally).   I feel like that could be tricky to make sure that it does get loaded.

Reply:
> The answer prefix is clearly not being set (or not set uniquely) for the problems when they are loaded for problem sets

I'm not even getting the problems to fetch from the server... it doesn't look like there's any implementation of `fetchSetProblems` anywhere in the code?

Reply:
Somehow I ended up missing this line during a merge... `fetchSetProblems` -- fixed now.

Reply:
opened a PR on pstaab/webwork3#22 to fix mathquill on set details page